.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+# find the lines in the diff that contain +/- CONFIG
+# ignore the lines that contain @
+def find_matching_lines(input_string):
+    pattern = r'(?!.*@)[+-]\s*.*CONFIG.*'
```

Comment:
>r'(?!.*@)[+-]\s*.*CONFIG.*' [](http://example.com/codeflow?start=14&length=27)

What are we searching for exactly? If we only care about new entries, then I'm thinking we might be able to simplify the pattern. How about `^\+CONFIG_.*`?

Reply:
So this regex is looking for lines that do not have @ and do have + or - and CONFIG. I will look into expanding this further and expanding my regex powers : P

Reply:
Discussion note: finding configs and stripping unnecessary parts with a single regex.